packages/address-book-controller/src/AddressBookController.ts

```diff
@@ -29,7 +45,9 @@ export interface ContactEntry {
  * AddressBookEntry representation
  * @property address - Hex address of a recipient account
  * @property name - Nickname associated with this address
- * @property chainId - Chain id identifies the current chain
+ * @property chainId - The id of a chain as per EIP-155, expressed as a decimal,
+ * encoded as a string. Note that the extension (and sometimes the mobile app)
+ * calls this the "network id".
```

Comment:
Do we call it the network ID?

There is a separate thing called the network ID that we still use in various places, but it's deprecated. I'm not aware of any examples of us calling the chain ID the "network ID".

Reply:
It's a little confusing.

In the extension, we seem to have both network ID and chain ID. Here is the definition of a network ID: https://github.com/MetaMask/metamask-extension/blob/7e7be2769df7c24413dc6ac92404a31ad2d112f2/shared/constants/network.ts#L179 And here is one place where it still seems to be getting used: https://github.com/MetaMask/metamask-extension/blob/7e7be2769df7c24413dc6ac92404a31ad2d112f2/ui/pages/confirm-transaction-base/confirm-transaction-base.component.js#L771

In mobile, chain id seems to be the same thing as network ID. For example: https://github.com/MetaMask/metamask-mobile/blob/main/app/util/networks/index.js#L38 And here is where it's being referenced: https://github.com/MetaMask/metamask-mobile/blob/main/app/core/NotificationManager.js#L411

Reply:
Hmm. I am a bit confused then. Given that we use `net_version` as a fallback for what I call the "original chain ID" in the case where an RPC entry didn't have a chain ID and now it does, I was under the impression that `net_version` returned a decimal version of the chain ID. This is what the extension refers to as the "network ID", so I connected those dots.

I was already confused, because what the AddressBookController calls the "chain ID" here isn't the chain ID — or at least it isn't in the same format as the chain ID that you get from `eth_chainId`.

Reply:
Regardless I will remove this comment so as not to create more confusion.

Reply:
> Given that we use `net_version` as a fallback for what I call the "original chain ID" in the case where an RPC entry didn't have a chain ID and now it does, I was under the impression that `net_version` returned a decimal version of the chain ID

It does not; they are totally distinct concepts. We only used `net_version` as a fallback for older networks that didn't yet support `eth_chainId`. We have since introduced `eth_chainId` as a requirement (quite a while ago), so remaining uses of `net_version` as a fallback are just technical debt and should be removed.

>what the AddressBookController calls the "chain ID" here isn't the chain ID — or at least it isn't in the same format

It is most definitely the same ID. Just a different format.

Reply:
I find this comment thread is insightful and educative beyond the scope of this PR. This distinction is important but I have a feeling most contributors are not aware of the above. Is this documented somewhere already (beyond EVM specs)?